features/steps/patroni_api.py

```diff
@@ -103,18 +103,35 @@ def scheduled_failover(context, at_url, from_host, to_host, in_seconds):
     """.format(at_url, from_host, to_host, datetime.now(pytz.utc) + timedelta(seconds=int(in_seconds))))
 
 
+@step('I issue a scheduled restart at {url:url} in {in_seconds:d} seconds with {data}')
+def scheduled_restart(context, url, in_seconds, data):
+    data = data and json.loads(data) or {}
+    restart_options = ['"schedule": "{0}"'.format((datetime.now(pytz.utc) + timedelta(seconds=int(in_seconds))).isoformat())]
+    for key in data:
+        if key == 'schedule':
+            continue
+        restart_options.append('"{0}": "{1}"'.format(key, data[key]))
```

Comment:
Could this for loop be replaced with something like `json.dumps(data.update(schedule=...))`?